byway/src/main/webapp/track-interests.js

```diff
@@ -0,0 +1,71 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+document.addEventListener('DOMContentLoaded', loadButtons());
```

Comment:
Two things:

1) See the note on the MDN page about how the page might be loaded before this script is executed, in which case you don't need to add an event listener.

2) `document.addEventListener('DOMContentLoaded', loadButtons());` Notice that what you're passing here is the *result* of calling `loadButtons`, you're not actually scheduling the execution of `loadButtons`.

Reply:
Fixed